src/drivers/bitbucket.go

```diff
@@ -15,8 +15,8 @@ type bitbucketCodeHostingDriver struct {
 	repository string
 }
 
-func (d *bitbucketCodeHostingDriver) CanBeUsed() bool {
-	return d.hostname == "bitbucket.org" || strings.Contains(d.hostname, "bitbucket")
+func (d *bitbucketCodeHostingDriver) CanBeUsed(driverOverride string) bool {
+	return driverOverride == "bitbucket" || d.hostname == "bitbucket.org"
```

Comment:
Are we sure we want to remove the auto-detection where if the hostname contains `bitbucket`, we assume it is a Bitbucket repository? I see very little downside to this, 99% of the time this will the right thing. Git Town should try to work well out of the box. The user should only have to configure things via manual config entries if Git Town cannot figure things out on its own.

Reply:
I don't think the auto-detection makes sense anymore. That was used for ssh-identities, but no longer works with the repo and pull-request commands. The support for self-hosted versions broke that as now it would use there ssh-identity name as the host.

Reply:
What if one doesn't use SSH identities, but a self-hosted Git server? Like `github-internal.deutsche-boerse.com`. It would be nice if Git Town works with that and is smart enough to figure out that this is a Github server.

Reply:
If we leave auto-detection in then it works out of the box for self-hosted versions that contain the word github, but would not for ssh identities that contain the name. I don't like having something be broken like that out of the box.

We should probably add the docs about how to make ssh identities and self hosted versions work to the unsupported service method so its faster for a user to get those set up.

Reply:
Okay, that's fair. In this case, we should add a step to the configuration wizard that if it sees a non-public origin URL, it asks the user about the server and adds the new config settings on its own.

Reply:
Sounds good. Can that be dealt with separately? I'd prefer not to add more to this PR.

Reply:
Yes absolutely! It **should** be in its own PR!